lib/queue.rb

```diff
@@ -0,0 +1,46 @@
+require_relative "attendee"
+require_relative "event_reporter"
+
+class List
+  attr_reader :queue
+
+  def initialize
+    @queue = []
+  end
+
+  def queue_count
+    @queue.length
+  end
+
+  def clear_queue
+    @queue.clear
+  end
+
+  def print_queue
+    @queue.each { |x| print x }
+  end
+
+  #this is NOT WORKINGGG
+  def print_sorted(attribute)
```

Comment:
sort_by needs a block passed to it. Something like:
```ruby
%w{ apple pear fig }.sort_by {|word| word.length}
              #=> ["fig", "pear", "apple"]
```

Reply:
In this case whatever is in the queue I'm guessing has either a key of the attribute or a method of the attribute you are passing in. So maybe:
```ruby
@queue.sort_by do |thing_in_queue|
  thing_in_queue[attribute] # or if it is a method thing_in_queue.send(attribute)
end
```